lib/improver/utilities/cube_manipulation.py

```diff
@@ -524,18 +524,48 @@ def _equalise_cell_methods(cubes):
     return cubelist
 
 
-def compare_attributes(cubes):
+def get_filtered_attributes(cube, attribute_filter=None):
+    """
+    Build dictionary of attributes that match the attribute_filter. If the
+    attribute_filter is None, return all attributes.
+
+    Args:
+        cube (iris.cube.Cube):
+            A cube from which attributes partially matching the
+            self.grid_metadata_identifier will be returned.
+    Keyword Args:
+        attribute_filter (string or None):
+            A string to match, or partially match, against attributes to build
+            a filtered attribute dictionary. If None, all attributes are
+            returned.
+    Returns:
+        attributes (dict):
+            A dictionary of attributes partially matching the attribute_filter
+            that were found on the input cube.
+    """
+    attributes = cube.attributes
+    if attribute_filter is not None:
+        attributes = {k: v for (k, v) in attributes.items()
+                      if attribute_filter in k}
+        return attributes
+
+    return dict(cube.attributes)
```

Comment:
Should this just be `attributes`? I would have thought that we could just have one return statement in this function i.e.
```python
     attributes = cube.attributes
     if attribute_filter is not None:
         attributes = {k: v for (k, v) in attributes.items()
                       if attribute_filter in k}
     return attributes
```

Reply:
Hmm, although now I test it I seem to be able to. This was definitely a problem before, so I wonder what I have changed...

Reply:
Ah, it was when I was still maintaining python2 compatibility. In that case I had to use the itervals() method, and that does not exist on the iris custom type. The items() method however does, so I can lose the dict() above as I have decided to abandon python2 and I will just have to generate the notebook figures using something other than moxie (or get the moxie python 3 version working).